### PR TITLE
Wwang talend/tdi 45481

### PR DESCRIPTION
**Pull request by wwang-talend**
### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant

### CODE REVIEW

File: component-api/src/main/java/org/talend/sdk/component/api/service/connection/CreateConnection.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import java.lang.annotation.Target;
+
+import org.talend.sdk.component.api.meta.Documentation;
+import org.talend.sdk.component.api.service.ActionType;
+
+@ActionType(value = "createconnection", expectedReturnedType = Object.class)
+@Target(METHOD)
+@Retention(RUNTIME)
+@Documentation("This class marks an action creating a connection")
+public @interface CreateConnection {
```
Comment (undx) [inline component-api/src/main/java/org/talend/sdk/component/api/service/connection/CreateConnection.java:31]: Extends documentation and specify that this is _studio only_
If this annotation is intended to be implemented in a service class, please add a some validation if in/out classes inherit RuntimeContextInjector.

**Review by undx — COMMENTED** [review]
Some comments and a question about this PR.

I add a quick look at TUP's PR (ain't finished apparently) and ...
What I don't get is how to you merge connection (datastore) configuration to connector's configuration?

Also, may need to add also a documentation (.adoc) in Studio documentation if some extra steps for enabling shared connection
is required.

Updated content.  adoc is ok

File: component-api/src/main/java/org/talend/sdk/component/api/context/RuntimeContextInjector.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+package org.talend.sdk.component.api.context;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class RuntimeContextInjector {
```
[the remark this answers is not in the document]
Comment (wwang-talend) [reply component-api/src/main/java/org/talend/sdk/component/api/context/RuntimeContextInjector.java:25]: yes, need to extends it

File: component-api/src/main/java/org/talend/sdk/component/api/service/connection/CreateConnection.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import org.talend.sdk.component.api.meta.Documentation;
+import org.talend.sdk.component.api.service.ActionType;
+
+@ActionType(value = "createconnection", expectedReturnedType = Object.class)
```
[the remark this answers is not in the document]
Comment (wwang-talend) [reply component-api/src/main/java/org/talend/sdk/component/api/service/connection/CreateConnection.java]: fixed

File: component-api/src/main/java/org/talend/sdk/component/api/service/connection/CloseConnection.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import org.talend.sdk.component.api.meta.Documentation;
+import org.talend.sdk.component.api.service.ActionType;
+
+@ActionType(value = "closeconnection", expectedReturnedType = CloseConnectionObject.class)
```
[the remark this answers is not in the document]
Comment (wwang-talend) [reply component-api/src/main/java/org/talend/sdk/component/api/service/connection/CloseConnection.java]: fixed

File: component-api/src/main/java/org/talend/sdk/component/api/context/RuntimeContextInjector.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (wwang-talend) [reply component-api/src/main/java/org/talend/sdk/component/api/context/RuntimeContextInjector.java:25]: the code generation which studio code for connection
```
final java.util.Map<String, String> configuration_NetSuiteConnection_1 = new java.util.HashMap<>();
				final java.util.Map<String, String> registry_metadata_NetSuiteConnection_1 = new java.util.HashMap<>();
				configuration_NetSuiteConnection_1.put("configuration.apiVersion", "V2019_2");
				configuration_NetSuiteConnection_1.put("configuration.account", "xxx");
				configuration_NetSuiteConnection_1.put("configuration.loginType", "BASIC");
				configuration_NetSuiteConnection_1.put("configuration.email", "xxx");
				configuration_NetSuiteConnection_1.put("configuration.password",
						routines.system.PasswordEncryptUtil.decryptPassword(
								"enc:routine.encryption.key.v1:xxxx"));
				configuration_NetSuiteConnection_1.put("configuration.role", "xxx");
				configuration_NetSuiteConnection_1.put("configuration.applicationId",
						"xxx");
				configuration_NetSuiteConnection_1.put("configuration.consumerKey", "");
				configuration_NetSuiteConnection_1.put("configuration.tokenId", "");
				configuration_NetSuiteConnection_1.put("configuration.__version", "-1");

				mgr_NetSuiteConnection_1.findPlugin("netsuite").get()
						.get(org.talend.sdk.component.runtime.manager.ContainerComponentRegistry.class).getServices()
						.stream().flatMap(c -> c.getActions().stream())
						.filter(actionMeta -> "createconnection".equals(actionMeta.getType())).forEach(actionMeta -> {
							Object connnection = actionMeta.getInvoker().apply(configuration_NetSuiteConnection_1);

							globalMap.put("conn_NetSuiteConnection_1", connnection);
						});

```

**Comment by wwang-talend** [comment]
have fixed all comments, will dicuss this with @undx :
```
If this annotation is intended to be implemented in a service class, please add a some validation if in/out classes inherit RuntimeContextInjector.
```

**Review by undx — APPROVED** [review]
LGTM

**Pull request merged**